cmd/cli/plugin/package/ui.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	uitable "github.com/cppforlife/go-cli-ui/ui/table"
+	"github.com/spf13/cobra"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/component"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/log"
+)
+
+type AdapterUI struct {
+	ui.WriterUI
+	outWriter io.Writer
+
+	outputFormat string
+	tableCount   int
+}
+
+// PrintLinef overrides go-cli-ui/ui.PrintLinef
+// It is used to print lines, but excludes certain lines based on the pattern
+func (adapterUI *AdapterUI) PrintLinef(pattern string, args ...interface{}) {
+	if strings.Contains(pattern, "Target cluster") {
+		return
+	}
+	message := fmt.Sprintf(pattern, args...)
+	_, err := fmt.Fprintln(adapterUI.outWriter, message)
+	if err != nil {
+		log.Error(err, "UI.PrintLinef failed (message='%s')", message)
+	}
+}
+
+func (adapterUI *AdapterUI) SetOutputFormat(outputFormat string) {
+	adapterUI.outputFormat = outputFormat
+}
+
+// PrintTable overrides go-cli-ui/ui.PrintTable
```

Comment:
Similar to `PrintLinef`, `PrintTable` is also used to print table in kctrl, but since the output needs to be a bit different in Tanzu CLI and there should also be options to select output format (like `json`), this method was added to "override" the behaviour of `PrintLinef`.